ports/esp32/Makefile

```diff
@@ -267,7 +276,8 @@ CFLAGS += -DMICROPY_ESP_IDF_4=1
 endif
 
 # this is what ESPIDF uses for c++ compilation
-CXXFLAGS = -std=gnu++11 $(CFLAGS_COMMON) $(INC) $(INC_ESPCOMP)
+# do not add INC_NEWLIB
+CXXFLAGS = -std=gnu++11 $(CFLAGS_COMMON) $(INC) $(INC_ESPCOMP) $(INC_XTENSA)
```

Comment:
Can `$(INC_NEWLIB)` be added at the end here?  If so then it would seem that the simplest fix is to just move the `$(INC_XTENSA)` and `$(INC_NEWLIB)` includes to the end of `$(INC_ESPCOMP)` (without introducing any new INC variables).  Maybe this is what you allude to in your top comment...?

Reply:
For v4 only that's correct. You would need an if statement there because the xtensa includes are only necessary for the v4 toolchain.

Reply:
I don't see a way to do this without at least one new INC_ variable as the CXXFLAGS needs to not include newlib.

Here's the simplest change I can come up with: https://github.com/jimmo/micropython/commit/978cce868e95b5a24e86efaaf0c04dce64e1f5f6
Tested on:
 - IDF3.3.1 (old toolchain)
 - IDF4.0-beta1 (old toolchain)
 - IDF4.0-beta1 (new toolchain from install.sh / export.sh)

Reply:
Ok, thanks for the explanation, and @jimmo for finding a simpler change.  I guess we can go with that change.  @mcdeoliveira do you want to update your PR here, or shall I pull in the commit from @jimmo?

Reply:
Go with @jimmo 's. Happy to see the issue resolved and tested on all versions of the toolchain.